File: SYP.Web/Modules/Email/EmailLogs/EmailLogsForm.cs
namespace SYP.Email.Forms;

[FormScript("Email.EmailLogs")]
[BasedOnRow(typeof(EmailLogsRow), CheckNames = true)]
public class EmailLogsForm
{
    [ReadOnly(true)]
    public long EmailQueueId { get; set; }

    [ReadOnly(true)]
    public string EmailSubject { get; set; }

    [ReadOnly(true)]
    public string ToAddress { get; set; }

    [ReadOnly(true)]
    public int Status { get; set; }

    [ReadOnly(true), TextAreaEditor(Rows = 3)]
    public string StatusMessage { get; set; }

    [ReadOnly(true), TextAreaEditor(Rows = 3)]
    public string SmtpResponse { get; set; }

    [ReadOnly(true)]
    public DateTime ProcessStartTime { get; set; }

    [ReadOnly(true)]
    public DateTime ProcessEndTime { get; set; }

    [ReadOnly(true)]
    public int Duration { get; set; }

    [ReadOnly(true)]
    public DateTime InsertDate { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Email/EmailLogsForm.ts
import { DateEditor, EnumEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";
import { EmailLogStatus } from "./EmailLogStatus";

export interface EmailLogsForm {
    EmailQueueId: StringEditor;
    EmailSubject: StringEditor;
    ToAddress: StringEditor;
    Status: EnumEditor;
    StatusMessage: TextAreaEditor;
    SmtpResponse: TextAreaEditor;
    ProcessStartTime: DateEditor;
    ProcessEndTime: DateEditor;
    Duration: IntegerEditor;
    InsertDate: DateEditor;
}

export class EmailLogsForm extends PrefixedContext {
    static readonly formKey = 'Email.EmailLogs';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmailLogsForm.init) {
            EmailLogsForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = TextAreaEditor;
            var w3 = DateEditor;
            var w4 = IntegerEditor;

            initFormType(EmailLogsForm, [
                'EmailQueueId', w0,
                'EmailSubject', w0,
                'ToAddress', w0,
                'Status', w1,
                'StatusMessage', w2,
                'SmtpResponse', w2,
                'ProcessStartTime', w3,
                'ProcessEndTime', w3,
                'Duration', w4,
                'InsertDate', w3
            ]);
        }
    }
}

[EmailLogStatus]; // referenced types